### ISSUE TITLE
op-batcher: issues recovering from an expired sequencing window

### ISSUE THREAD

**Issue by zhiqiangxu** [ethereum-optimism/optimism#13150]
Currently if the batch is already expired(the distance between safe head and unsafe head is more than one sequence window), op-batcher will still publish it, which will be dropped by op-node [here](https://github.com/ethereum-optimism/optimism/blob/9f84a4e3ac73fadafed4f61889357115632cdf8f/op-node/rollup/derive/batches.go#L98).

Then op-batcher will detect it [here](https://github.com/ethereum-optimism/optimism/blob/9f84a4e3ac73fadafed4f61889357115632cdf8f/op-batcher/batcher/channel_manager.go#L564) since the safe head isn't advanced.(this feature is introduced [here](https://github.com/ethereum-optimism/optimism/pull/12430))

Then op-batcher will re-publish the expired batch again...

In the end, op-batcher will never catch up with the sequencing window and the balance will be exhausted very quickly as it's constantly sending transactions.

Is there any document on how op-batcher is expected to catch up in this case?

**Comment by sebastianst** [comment]
Related https://github.com/ethereum-optimism/optimism/issues/11228

**Comment by emilianobonassi** [comment]
@zhiqiangxu workaround we've seen working in these situations are disabling span batches and setting very short max channel duration

**Comment by zhiqiangxu** [comment]
> [@zhiqiangxu](https://github.com/zhiqiangxu) workaround we've seen working in these situations are disabling span batches and setting very short max channel duration

Hey @emilianobonassi , thanks for the suggestion!

The trick still works for Holocene HF, but it needs to be done manually.

[Here](https://github.com/ethereum-optimism/optimism/pull/13308) is a PR trying to make it automatic.

**Comment by emilianobonassi** [comment]
yeah i know @zhiqiangxu - thanks for implementing the automation!

very short max channel duration is what helps here, so channels contains just a few batches and when discarded you minimize the drop/rewrites.

it would be great implementing a "recovery status" of the sequencer as mentioned https://github.com/ethereum-optimism/optimism/issues/11228 and a good feedback loop with the batcher. so its aware and doesnt post batches related to empty blocks that will be discarded anyway (during reorg safe head moves anyway even without reads)

in the past batcher was decoupled and very simple, now with holocene, given the simplification of the derivation pipeline, we should be a lil bit smart on how we write to minimize the probability of these situations.

**Comment by zhiqiangxu** [comment]
> it would be great implementing a "recovery status" of the sequencer as mentioned https://github.com/ethereum-optimism/optimism/issues/11228 and a good feedback loop with the batcher. so its aware and doesnt post batches related to empty blocks that will be discarded anyway (during reorg safe head moves anyway even without reads)

Understood, that would require a spec change though, which is not feasible at this point. The above pr might already be good enough in practice, and it's definitely better than current behavior: which will exhaust balance doing nothing if not interfered.

**Comment by emilianobonassi** [comment]
oh @zhiqiangxu - my was just a proposal not direct call to action, more for op team (@sebastianst et al) for the next iterations.
